doc/book/migration/rc-to-v1.md

```diff
@@ -0,0 +1,359 @@
+be undesired. As a result, we now require you to inject each manually.
+
+### Impact
+
+This change will require changes in your application.
+
+1. If you are using Expressive programmatically (i.e., you are not using
+   a container and the `Zend\Expressive\Container\ApplicationFactory`),
+   you are now *required* to call `Application::pipeRoutingMiddleware()`.
+   Additionally, a new method, `Application::pipeDispatchMiddleware()` exists
+   for injecting the application with the dispatch middleware, this, too, must
+   be called.
+
+   This has a fortunate side effect: registering routed middleware no longer
+   affects the middleware pipeline order. As such, you can register your
+   pipeline first or last prior to running the application. The only stipulation
+   is that _unless you register the routing **and** dispatch middleware, your routed
+   middleware will not be executed!_ As such, the following two lines **must**
+   be added to your application prior to calling `Application::run()`:
+
+```php
+$app->pipeRoutingMiddleware();
+$app->pipeDispatchMiddleware();
+```
+
+2. If you are creating your `Application` instance using a container and the
+   `Zend\Expressive\Container\ApplicationFactory`, you will need to update your
+   configuration to list the routing and dispatch middleware. The next section
+   details the configuration changes necessary.
+
+## ApplicationFactory configuration changes
+
+As noted in the document summary, the middleware pipeline configuration was
+changed starting in RC6.  The changes are done in such a way as to honor
+configuration from RC5 and earlier, but using such configuration will now prompt
+you to update your application.
+
+RC5 and earlier defined the default `middleware_pipeline` configuration as follows:
+
+```php
+return [
+    'middleware_pipeline' => [
+        // An array of middleware to register prior to registration of the
+        // routing middleware
+        'pre_routing' => [
+            //[
+            // Required:
+            //    'middleware' => 'Name or array of names of middleware services and/or callables',
+            // Optional:
+            //    'path'  => '/path/to/match',
+            //    'error' => true,
+            //],
+            [
+                'middleware' => [
+                    Helper\ServerUrlMiddleware::class,
+                    Helper\UrlHelperMiddleware::class,
+                ],
+            ],
+        ],
+
+        // An array of middleware to register after registration of the
+        // routing middleware
+        'post_routing' => [
+            //[
+            // Required:
+            //    'middleware' => 'Name of middleware service, or a callable',
+            // Optional:
+            //    'path'  => '/path/to/match',
+            //    'error' => true,
+            //],
+        ],
+    ],
+]
+```
+
+### Impact
+
+While the configuration from RC5 and earlier will continue to work, it will
```

Comment:
No. If we did that, we'd need another RC between RC6 and stable that removes the functionality. Additionally, I know a number of folks are already pushing Expressive applications to production, so giving them until 1.1 ensures they have adequate time to upgrade.